File: src/main/java/ch/itds/taglib/asset/pipeline/thymeleaf/AssetSrcAttributeTagProcessor.java
package ch.itds.taglib.asset.pipeline.thymeleaf;

import org.thymeleaf.templatemode.TemplateMode;

public class AssetSrcAttributeTagProcessor extends AbstractAssetAttributeTagProcessor {

    private static final String ATTR_NAME = "src";
    private static final int PRECEDENCE = 1000;

    public AssetSrcAttributeTagProcessor(final String dialectPrefix) {
        super(
                TemplateMode.HTML, // This processor will apply only to HTML mode
                dialectPrefix,     // Prefix to be applied to name for matching
                null,              // No tag name: match any tag name
                false,             // No prefix to be applied to tag name
                ATTR_NAME,         // Name of the attribute that will be matched
                true,              // Apply dialect prefix to attribute name
                PRECEDENCE,        // Precedence (inside dialect's precedence)
                true);             // Remove the matched attribute afterwards
    }
}

File: src/main/java/com/dra11y/assetThymeleafTaglib/processor/AssetSrcAttributeTagProcessor.kt
package com.dra11y.assetThymeleafTaglib.processor

import com.dra11y.assetThymeleafTaglib.asset.AssetResolver
import org.springframework.stereotype.Component
import org.thymeleaf.templatemode.TemplateMode

@Component
class AssetSrcAttributeTagProcessor(dialectPrefix: String?, assetResolver: AssetResolver) : AbstractAssetAttributeTagProcessor(
    TemplateMode.HTML, // This processor will apply only to HTML mode
    dialectPrefix, // Prefix to be applied to name for matching
    null, // No tag name: match any tag name
    false, // No prefix to be applied to tag name
    ATTR_NAME, // Name of the attribute that will be matched
    true, // Apply dialect prefix to attribute name
    PRECEDENCE, // Precedence (inside dialect's precedence)
    true,
    assetResolver
) {
    init {
        println("init AssetSrcAttributeTagProcessor")
    }

    companion object {
        private const val ATTR_NAME = "src"
        private const val PRECEDENCE = 1000
    }
}
